### ISSUE TITLE
Published artifacts contain invalid bytecode

### ISSUE THREAD

**Issue by ivoadf** [OpenHFT/Chronicle-Network#227]
A javac bug similar to [this one](https://bugs.openjdk.org/browse/JDK-8187805) is resulting invalid bytecode for class `AbstractStatelessClient.java` that is being included in the published artifacts

From this issue https://github.com/gradle/gradle/issues/23405 :

> It looks like the library in question is hitting some javac bug similar to [this one](https://bugs.openjdk.org/browse/JDK-8187805), and the produced bytecode is in fact invalid. It can be fixed by switching to a more recent JDK (e.g. 11) and compiling the library with -release 8 to maintain the binary compatibility - but this has to be done by the library maintainers.

**Comment by alamar** [comment]
Can you please share the specific indication of this issue?

**Comment by ivoadf** [comment]
@alamar  The specific indication can be found in the gradle issue I linked https://github.com/gradle/gradle/issues/23405 
When processing the ChronicleNetwork jar gradle throws a ArrayIndexOutOfBoundsException according to the gradle team this is due to the bytecode for `AbstractStatelessClient.java` being invalid (refer to [this comment](https://github.com/gradle/gradle/issues/23405#issuecomment-1378798200))

**Comment by alamar** [comment]
Can you please list steps to reproduce?

**Comment by ivoadf** [comment]
> Can you please list steps to reproduce?

[Adding ChronicleNetwork as a buildScript dependency](https://github.com/ivoadf/gradle-bug-reproducer/blob/kotlin-dsl/build.gradle.kts#L6) reproduces the issue.

Here is the gradle output: https://github.com/ivoadf/gradle-bug-reproducer/actions/runs/3909828510/jobs/6681335514#step:4:491

**Comment by alamar** [comment]
Is there a way to run this script from the command line without knowledge how to operate gradle? Which is lacking unfortunately.

**Comment by NikolayMetchev** [comment]
The solution appears to be to compile with a newer compiler. Is that even a possibility at this stage?

**Comment by alamar** [comment]
A lot of things are possible but we need to be able to test our solutions against reproducer, and I can't figure out how to run a sole gradle script yet.

**Comment by mlopatkin** [comment]
FWIW, I suspect the issue in question is also the reason for [this workaround for jacoco](https://github.com/OpenHFT/Chronicle-Network/blob/ea/pom.xml#L365-L367). May be reproducing it with jacoco would be easier.

**Comment by ivoadf** [comment]
> A lot of things are possible but we need to be able to test our solutions against reproducer, and I can't figure out how to run a sole gradle script yet.

If you clone the `kotlin-dsl` branch https://github.com/ivoadf/gradle-bug-reproducer/tree/kotlin-dsl you should be able to reproduce locally from the command line by running `gradlew help --info --stacktrace`

**Comment by alamar** [comment]
```% gradle help --info --stacktrace                                                                                                                13:33
Initialized native services in: /home/ilyak/.gradle/native
The client will now receive all logging from the daemon (pid: 44630). The daemon log file: /home/ilyak/.gradle/daemon/4.4.1/daemon-44630.out.log
Starting 5th build in daemon [uptime: 58 mins 5.989 secs, performance: 96%]
Using 12 worker leases.
Creating new cache for fileHashes, path /home/ilyak/w/scratch/gradle-bug-reproducer/.gradle/4.4.1/fileHashes/fileHashes.bin, access org.gradle.cache.internal.DefaultCacheAccess@59e34590
Starting Build
Compiling settings file '/home/ilyak/w/scratch/gradle-bug-reproducer/settings.gradle' using SubsetScriptTransformer.
Compiling settings file '/home/ilyak/w/scratch/gradle-bug-reproducer/settings.gradle' using BuildScriptTransformer.
Settings evaluated using settings file '/home/ilyak/w/scratch/gradle-bug-reproducer/settings.gradle'.
Projects loaded. Root project using build file '/home/ilyak/w/scratch/gradle-bug-reproducer/build.gradle'.
Included projects: [root project 'reproducer-project']

> Configure project : 
Evaluating root project 'reproducer-project' using build file '/home/ilyak/w/scratch/gradle-bug-reproducer/build.gradle'.

All projects evaluated.
Selected primary task 'help' from project :
Tasks to be executed: [task ':help']
Creating new cache for resourceHashesCache, path /home/ilyak/w/scratch/gradle-bug-reproducer/.gradle/4.4.1/fileHashes/resourceHashesCache.bin, access org.gradle.cache.internal.DefaultCacheAccess@59e34590
Creating new cache for taskHistory, path /home/ilyak/w/scratch/gradle-bug-reproducer/.gradle/4.4.1/taskHistory/taskHistory.bin, access org.gradle.cache.internal.DefaultCacheAccess@7bdc8bbe
Creating new cache for outputFiles, path /home/ilyak/w/scratch/gradle-bug-reproducer/.gradle/buildOutputCleanup/outputFiles.bin, access org.gradle.cache.internal.DefaultCacheAccess@1f85e6f5
:help (Thread[Task worker for ':',5,main]) started.

> Task :help 
Putting task artifact state for task ':help' into context took 0.0 secs.
Up-to-date check for task ':help' took 0.0 secs. It is not up-to-date because:
  Task has not declared any outputs.

Welcome to Gradle 4.4.1.

To run a build, run gradle <task> ...

To see a list of available tasks, run gradle tasks

To see a list of command-line options, run gradle --help

To see more detail about a task, run gradle help --task <task>

:help (Thread[Task worker for ':',5,main]) completed. Took 0.001 secs.

BUILD SUCCESSFUL in 0s
1 actionable task: 1 executed
```
I'm not sure what's gradlew - not apt-gettable

**Comment by alamar** [comment]
```
% ./gradlew help --info --stacktrace
Downloading https://services.gradle.org/distributions/gradle-7.3-bin.zip
...........10%...........20%...........30%...........40%...........50%...........60%...........70%...........80%...........90%...........100%
Initialized native services in: /home/ilyak/.gradle/native
Initialized jansi services in: /home/ilyak/.gradle/native

Welcome to Gradle 7.3!

Here are the highlights of this release:
 - Easily declare new test suites in Java projects
 - Support for Java 17
 - Support for Scala 3

For more details see https://docs.gradle.org/7.3/release-notes.html

Removing 0 daemon stop events from registry
Starting a Gradle Daemon (subsequent builds will be faster)
Starting process 'Gradle build daemon'. Working directory: /home/ilyak/.gradle/daemon/7.3 Command: /usr/lib/jvm/java-8-openjdk-amd64/bin/java -XX:MaxMetaspaceSize=256m -XX:+HeapDumpOnOutOfMemoryError -Xms256m -Xmx512m -Dfile.encoding=UTF-8 -Duser.country=RU -Duser.language=ru -Duser.variant -cp /home/ilyak/.gradle/wrapper/dists/gradle-7.3-bin/ddwl0k7mt9g6ak16i1m905vyv/gradle-7.3/lib/gradle-launcher-7.3.jar org.gradle.launcher.daemon.bootstrap.GradleDaemon 7.3
Successfully started process 'Gradle build daemon'
An attempt to start the daemon took 0.725 secs.
The client will now receive all logging from the daemon (pid: 55318). The daemon log file: /home/ilyak/.gradle/daemon/7.3/daemon-55318.out.log
Starting build in new daemon [memory: 455.5 MiB]
Using 12 worker leases.
Watching the file system is configured to be enabled if available
Now considering [/home/ilyak/w/scratch/gradle-bug-reproducer] as hierarchies to watch
File system watching is active
Starting Build
Compiling settings file '/home/ilyak/w/scratch/gradle-bug-reproducer/settings.gradle' using SubsetScriptTransformer.
Compiling settings file '/home/ilyak/w/scratch/gradle-bug-reproducer/settings.gradle' using BuildScriptTransformer.
Settings evaluated using settings file '/home/ilyak/w/scratch/gradle-bug-reproducer/settings.gradle'.
Projects loaded. Root project using build file '/home/ilyak/w/scratch/gradle-bug-reproducer/build.gradle'.
Included projects: [root project 'reproducer-project']

> Configure project :
Evaluating root project 'reproducer-project' using build file '/home/ilyak/w/scratch/gradle-bug-reproducer/build.gradle'.
All projects evaluated.
Selected primary task 'help' from project :
Tasks to be executed: [task ':help']
Tasks that were excluded: []
:help (Thread[Execution worker for ':' Thread 11,5,main]) started.

> Task :help
Caching disabled for task ':help' because:
  Build cache is disabled
Task ':help' is not up-to-date because:
  Task has not declared any outputs despite executing actions.

Welcome to Gradle 7.3.

To run a build, run gradlew <task> ...

To see a list of available tasks, run gradlew tasks

To see more detail about a task, run gradlew help --task <task>

To see a list of command-line options, run gradlew --help

For more detail on using Gradle, see https://docs.gradle.org/7.3/userguide/command_line_interface.html

For troubleshooting, visit https://help.gradle.org
:help (Thread[Execution worker for ':' Thread 11,5,main]) completed. Took 0.087 secs.

BUILD SUCCESSFUL in 14s
1 actionable task: 1 executed
```

**Comment by ivoadf** [comment]
@alamar are you on the `kotlin-dsl` branch?

**Comment by alamar** [comment]
Thank you, I can see the problem now - indeed, forgot to switch branch.

**Comment by alamar** [comment]
Related: #59

**Comment by hft-team-city** [comment]
Released in [Chronicle-Network-2.24ea5](https://github.com/OpenHFT/Chronicle-Network/releases/chronicle-network-2.24ea5), [BOM-2.24ea29](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.24ea29)
